### PR TITLE
Implement Map Type

### PR DESCRIPTION
**Pull request by maikkundev**
# Resolves

This pull request resolves issue #46.

### CODE REVIEW

File: scenes/levels/level-type.gd
Comment (actuallylost) [inline scenes/levels/level-type.gd:1]: Can you also add some variables that give details about a map i.e have one be the array of enemies the map has, one that sores the boss's name (keep them empty for now), and a boolean that's "unlocked".

**Review by actuallylost — CHANGES_REQUESTED** [review]

File: scenes/levels/level-type.gd [hunk 1]
```diff
@@ -5,3 +5,12 @@ extends Resource
 
 # Name of the level
 @export var name: String
+@export var unlocked: bool
+
+@export_group("Enemies")
+
+# Array of enemy types the level will have
+@export var enemies: Array
```
[the remark this answers is not in the document]
Comment (maikkundev) [reply scenes/levels/level-type.gd:13]: Our class doesn't inherit any Node-derived class, so this is not possible unless we change the inheritage of the script.
`Node export is only supported in Node-derived classes, but the current class inherits "Resource"`

[the remark this answers is not in the document]
Comment (actuallylost) [reply scenes/levels/level-type.gd:13]: Very well, set it to Entity if you can then

[the remark this answers is not in the document]
Comment (maikkundev) [reply scenes/levels/level-type.gd:13]: Nope, same issue. I say we leave it as is for now.

**Review by actuallylost — APPROVED** [review]
LGTM

**Pull request merged**